### PR TITLE
Sprint upgrade staging

### PR DESCRIPTION
**Pull request by rmathew1011**
Upgrade done for GIFMService

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,16 +1,23 @@
 server.port: 9001
 server.contextPath:
 security.basic.enabled: false
+
+# Bean overriding is necessary for TokenAuthorizationFilter.
+spring.main.allow-bean-definition-overriding: true
```
Comment (wwelling) [inline src/main/resources/application.properties:6]: For these required overrides, it would be best we have an issue created to identify the problem.

**Review by wwelling — COMMENTED** [review]
We should postpone merge into main until we finalize the Dockerfile standard we wish to have across the deployments.

File: src/main/resources/application.properties (hunk 1 above)
Comment (rmathew1011) [reply src/main/resources/application.properties:6]: Will create an issue for the same

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
+# Circular bean dependency between appWebMvcConfig and userRepoImpl.
+spring.main.allow-circular-references: true
```
[the remark this answers is not in the document]
Comment (rmathew1011) [reply src/main/resources/application.properties:8]: Issue created for: https://github.com/TAMULib/GIFMButtonService/issues/45

**Comment by rmathew1011** [comment]
> We should postpone merge into main until we finalize the Dockerfile standard we wish to have across the deployments.

Dockerfile updated as per  the standard

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+USER $USER_NAME
+
+# Set deployment directory.
+WORKDIR $HOME_DIR
+
+# Copy over the built artifact from the maven image.
+COPY --from=maven $SOURCE_DIR/target/ROOT.jar ./sage.jar
+
+# Run java command.
+CMD ["java", "-jar", "/GIFMButtonService/ROOT.jar"]
```
Comment (wwelling) [inline Dockerfile:62]: This is not correct. The COPY places the jar at `./sage.jar`.

Comment (rmathew1011) [reply Dockerfile:62]: Made this correction

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# Settings.
+ARG USER_ID=3001
+ARG USER_NAME=gifmbutton-service
```
Comment (wwelling) [inline Dockerfile:3]: Why are we putting service in the username? Can we just use gifm or something simple?

File: pom.xml [hunk 5]
```diff
@@ -76,19 +91,29 @@
...
         <configuration>
           <failOnMissingWebXml>false</failOnMissingWebXml>
-          <release>11</release>
+          <release>${java.version}</release>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.22.2</version>
```
Comment (wwelling) [inline pom.xml:110]: Lets move these versions to properties for consistency with other services.

```
<maven-plugins.version>2.22.2</maven-plugins.version>
```

File: src/main/java/edu/tamu/app/config/AppWebMvcConfig.java [hunk 6]
```diff
@@ -59,7 +59,6 @@ public ConfigurableMimeFileTypeMap configurableMimeFileTypeMap() {
         return new ConfigurableMimeFileTypeMap();
     }
 
-    @Override
```
Comment (wwelling) [inline src/main/java/edu/tamu/app/config/AppWebMvcConfig.java:62]: I know Java 11 doesn't require the override annotation it does help to know.

File: src/main/java/edu/tamu/app/service/SfxService.java [hunk 7]
```diff
@@ -18,17 +17,21 @@
...
 /**
  * The SfxService checks with the linkresolver to find out if the full text is available for the given url
  *
  * @author Jason Savell <jsavell@library.tamu.edu>
  *
  */
-
-
 @Service
+@EnableConfigurationProperties(AppConfig.class)
```
Comment (wwelling) [inline src/main/java/edu/tamu/app/service/SfxService.java:29]: Please move `@EnableConfigurationProperties` to WebServerInit or an applicable config file. It is only required once, and it appears on multiple services.

**Review by jsavell — APPROVED** [review]
Testing only  with `mvn spring-boot:run`, all functions in the UI and public API seem to be working as expected.

**Review by wwelling — APPROVED** [review]

**Pull request merged**